Mod1.Lection2.Hw1/Mod2.Lection2.Hw1/Repositories/CartRepository.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Mod2.Lection2.Hw1.Models;
+
+namespace Mod2.Lection2.Hw1.Repositories;
+
+internal class CartRepository
+{
+    private readonly Product _product; // В конструкторе принимаем заполненный экземпляр Product
+
+    public CartRepository(Product product)
+    {
+        _product = product; // Сохраняем переданный экземпляр Product
+    }
+
+    public void AddtoCart(Cart cart)
+    {
+        while (true)
+        {
+            cart.SelectedProduct = Console.ReadLine(); // Здесь устанавливаем SelectedProduct через свойство
+
+            if (cart.SelectedProduct != null && _product.dictionaryProductPrice.TryGetValue(cart.SelectedProduct, out var price))
+            {
+                if (cart.CartProducts.Count >= 10)
```

Comment:
Validation belongs to the service. Here you just store data and check related logic (like whether you have enough space to store, or if item already stored)